public/scripts/world-info.js

```diff
@@ -3269,12 +3277,13 @@ async function _save(name, data) {
     eventSource.emit(event_types.WORLDINFO_UPDATED, name, data);
 }
 
-async function saveWorldInfo(name, data, immediately) {
+async function saveWorldInfo(name, data, immediately = false) {
     if (!name || !data) {
         return;
     }
 
-    worldInfoCache.delete(name);
+    // Update cache immediately, so any future call can pull from this
+    worldInfoCache.set(name, data);
```

Comment:
I just realized that every single call to saveWorldInfo would create a new clone of the entire data structure. There are at least 10 calls per entry. With a max of 1000 entries per page. Incredibly wasteful memorywise. I'm thinking of moving this into _save instead.

Reply:
You said clone, not me :P 
But Yes, you are right. Haven't thought about that.

You can't move it into the denounce though. The next call pulling from the data and saving another field might already happen, and it needs the modified object.
Think about habe an STscript that sets three different fields.

Maybe using deep clones isn't the way to go here. Maybe it shouldn't clone on set, and we just hope no one is dumb enough to modify the data object after the save.

Reply:
I think it's fine to put an original object on set, but clone it on get. Either way, we'll get a copy from the cache that won't interfere with the editor.